### ISSUE TITLE
MA0101 is reported for raw string literals

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#631]
### Version of the Meziantou.Analyzer NuGet package

2.0.100

### Rule Identifier

MA0101

### Target Framework

net7.0

### C# Language version

C# 11

### Description

While not wrong, I still wonder if **MA0101** should be reported for a raw string literal that spans multiple lines.
Personally, I'm not 100% happy with the fact that the line ending depends on the line ending of the source file, but there are no better (easy readable) alternatives.

Alternative would be to have an option that controls whether **MA0101** also applies to raw string literals.

### Reproduction Steps

**Minimal code:**
```c#
var x = $$$"""
        This should be
        OK!
        Right?
        """;

Console.WriteLine(x);
```

**Expected result:**
MA0101 is not reported.

**Actual result:**
MA0101: String contains an implicit end of line character (https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0101.md)

**Comment by meziantou** [comment]
I think there should be an option. Also, the current code is buggy for some cases.
